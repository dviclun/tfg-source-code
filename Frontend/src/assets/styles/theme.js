import { createTheme } from "@mui/material";
import { orange } from '@mui/material/colors';
import Poppins from '../fonts/Poppins-Regular.ttf'

export const theme = createTheme({
    status: {
        danger: orange[500],
      },
    palette: {
        slate100: '#f1f5f9',
        slate300:'#cbd5e1',
        green300: '#86efac',
        green400: '#4ade80',
        green500: '#22c55e',
        green600: '#16a34a',
        green700: '#047857', 
        green900: '#14532d',
        red400: '#f87171',
        red500: '#ef4444',
        zinc200: '#e4e4e7',
        zinc300: '#d4d4d8',
        zinc400: '#a1a1aa',
        rose400: '#fb7185',
        rose500: '#f43f5e',
        purple400: '#c084fc',
        purple500: '#a855f7',
        blue300: '#93c5fd',
        blue400: '#60a5fa',
        gray100: '#f3f4f6',
        neutral100: '#f5f5f5',
        neutral200: '#e5e5e5',
        emerald100: '#d1fae5'
    },
});