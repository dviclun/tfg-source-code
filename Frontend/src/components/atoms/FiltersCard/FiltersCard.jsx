import { Card, CardActionArea, CardContent, CardMedia, Typography, useMediaQuery } from '@mui/material'
import { useTheme } from "@mui/material/styles";
import PropTypes from 'prop-types';

import React from 'react'


export const FiltersCard = ({ cardTitle, imgHeight, muscularGroup, groupId, handleFilterClick, loadImage: loadImageProp }) => {

    const theme = useTheme();

    const isLarge = useMediaQuery(theme.breakpoints.down('lg'));

    const loadImage = loadImageProp || (() => {
        switch (muscularGroup.toLowerCase()) {
            case 'abs':
                return 'https://i.imgur.com/CZoXamC.jpg';

            case 'arms':
                return 'https://i.imgur.com/0cxsnnW.jpeg';

            case 'back':
                return 'https://i.imgur.com/6ha1HRo.jpeg';

            case 'biceps':
                return 'https://i.imgur.com/V7FmUoe.jpeg';
            case 'calves':
                return 'https://i.imgur.com/uzsr0mq.jpeg';
            case 'chest':
                return 'https://i.imgur.com/QkAeB2m.jpeg';
            case 'forearms':
                return 'https://i.imgur.com/1kwqMrd.jpeg';
            case 'glutes':
                return 'https://i.imgur.com/Vg6s5th.jpeg';
            case 'hamstrings':
                return 'https://i.imgur.com/ZpJS8lp.jpeg';
            case 'legs':
                return 'https://i.imgur.com/LvwURJZ.jpeg';
            case 'quadriceps':
                return 'https://i.imgur.com/GcxQJCD.jpeg';
            case 'shoulders':
                return 'https://i.imgur.com/TKyajtc.jpeg';
            case 'triceps':
                return 'https://i.imgur.com/qJwE6hn.jpeg';
            default:
                return 'https://i.imgur.com/CZoXamC.jpg'
        }
    })

    return (
        <Card className="trainingCard">
            <CardActionArea onClick={() => handleFilterClick(groupId)}>
                <CardMedia
                    className='cardImage'
                    component="img"
                    height={imgHeight}
                    image={loadImage()}
                />
                <CardContent className='cardBg' sx={{ position: "absolute", top: 0, zIndex: '100', width: '100%', height: '100%', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
                    <Typography variant={isLarge ? 'body2' : 'h6'} sx={{ fontFamily: 'Poppins', fontWeight: 'bold', color: '#fff', fontStyle: 'italic' }}>
                        {cardTitle}
                    </Typography>
                </CardContent>
            </CardActionArea>
        </Card>
    )
}

FiltersCard.propTypes = {
    cardTitle: PropTypes.string,
    imgHeight: PropTypes.string,
    muscularGroup: PropTypes.string,
    groupId: PropTypes.number,
    handleFilterClick: PropTypes.func
}