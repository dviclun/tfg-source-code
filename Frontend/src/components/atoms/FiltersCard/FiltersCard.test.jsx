import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { AuthProvider } from "../../../contexts/Auth/AuthProvider";
import { ThemeProvider } from "@emotion/react";
import { theme } from "../../../assets/styles/theme";
import i18next from "i18next";
import { I18nextProvider } from "react-i18next";
import { FiltersCard } from "./FiltersCard";

describe('Test in <FiltersCard/>', () => { 

    const mockLoadImage = jest.fn();

    test('Container must render and match snapshot', () => { 
        mockLoadImage.mockReturnValue('');
        const {container} = render(
            <BrowserRouter>
                <AuthProvider>
                    <ThemeProvider theme={theme}>
                    <I18nextProvider i18n={i18next}>
                        <FiltersCard loadImage={mockLoadImage}/>    
                    </I18nextProvider>
                    </ThemeProvider>
                </AuthProvider>
            </BrowserRouter>
        )

        expect(container).toBeTruthy();
        expect(container).toMatchSnapshot();
     })
})