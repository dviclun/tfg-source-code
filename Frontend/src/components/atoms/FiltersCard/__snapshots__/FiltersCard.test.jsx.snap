// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FiltersCard/> Container must render and match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root trainingCard css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <img
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img cardImage css-o69gx8-MuiCardMedia-root"
      />
      <div
        class="MuiCardContent-root cardBg css-19akuni-MuiCardContent-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-fy86or-MuiTypography-root"
        />
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
